platforms/core-configuration/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt

```diff
@@ -59,4 +73,48 @@ abstract class KotlinDslPlugin : Plugin<Project> {
             )
         }
     }
+
+    private
+    fun Project.workaroundKotlinJavaVersionSupport() {
+        val targetJavaVersion = deprecatedJvmTargetVersion
+            .orElse(defaultToolchainJavaVersion)
+            .map(::toLastSupportedKotlinJavaVersion)
+        tasks.withType(KotlinCompile::class.java).configureEach { kotlinCompile ->
+            kotlinCompile.compilerOptions.jvmTarget.set(
+                targetJavaVersion.map { JvmTarget.fromTarget(it.toString()) }
+            )
+        }
+        tasks.withType(JavaCompile::class.java).configureEach { javaCompile ->
+            javaCompile.targetCompatibility = targetJavaVersion.get().toString()
+        }
+    }
+
+    private
+    val Project.deprecatedJvmTargetVersion: Provider<JavaVersion>
```

Comment:
I would say don't call this "deprecated...", it's very confusing. I now know what you mean, but took me a while.

Reply:
Renamed